forc-plugins/forc-crypto/src/keygen/parse_secret.rs

```diff
@@ -0,0 +1,56 @@
+//! This file will be hosted here until
+//! https://github.com/FuelLabs/sway/issues/5170 is fixed
+use super::KeyType;
+use anyhow::Result;
+use fuel_core_types::{fuel_crypto::SecretKey, fuel_tx::Input};
+use libp2p_identity::{secp256k1, Keypair, PeerId};
+use serde_json::json;
+use std::{ops::Deref, str::FromStr};
+
+/// Parse a secret key to view the associated public key
+#[derive(Debug, clap::Args)]
+#[clap(
+    author,
+    version,
+    about = "Parses a private key to view the associated public key"
+)]
+pub struct Arg {
+    /// A private key in hex format
+    secret: String,
+    /// Key type to generate. It can either be `block-production` or `p2p`.
```

Comment:
Same issue here.

Reply:
My understanding is that peer_id and addresses are derived differently. Technically I _think_ a private key can be either an address or a peer_id.